docs: update testing examples to use Vitest instead of Jasmine

AryanBagade opened the pull request:
Replace Jasmine-specific APIs with Vitest equivalents in service testing examples

Fixes #65987

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #65987
The testing documentation examples use Jasmine-specific APIs (DoneFn, jasmine.createSpyObj, .withContext()) which don't work with Vitest. This causes confusion for developers trying to follow the Angular 21 testing guide, as Angular has migrated to Vitest.


## What is the new behavior?
The testing examples now use proper Vitest syntax:
  - Async tests use `async/await` instead of `done` callbacks
  - Mock creation uses `vi.fn().mockReturnValue()` instead of `jasmine.createSpyObj`
  - Custom error messages use Vitest's second parameter: `expect(value, 'message')` instead of `.withContext()`
  - Added proper `import {vi} from 'vitest'` statement


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information
This change helps complete the Vitest migration mentioned in the documentation. The examples now match the official Vitest documentation and will work correctly when developers copy them.

AryanBagade committed:
docs: update testing examples to use Vitest instead of Jasmine

Replace Jasmine-specific APIs with Vitest equivalents in service testing examples

Fixes #65987

JeanMeche commented on adev/src/content/examples/testing/src/app/demo/demo.spec.ts:
```diff
@@ -19,18 +20,18 @@ describe('demo (no TestBed):', () => {
+        service.getObservableValue().subscribe((val) => {
+          resolve(val);
+        });
       });
```
Should we maybe simply by using `firstValueFrom` instead?

JeanMeche commented on adev/src/content/examples/testing/src/app/demo/demo.spec.ts:
```diff
@@ -58,19 +59,16 @@ describe('demo (no TestBed):', () => {
+      };
 
-      masterService = new MasterService(valueServiceSpy);
+      masterService = new MasterService(valueServiceSpy as any);
```
as this point we shouldn't be introducing new `any`.

JeanMeche commented on adev/src/content/examples/testing/src/app/demo/demo.spec.ts:
```diff
@@ -58,19 +59,16 @@ describe('demo (no TestBed):', () => {
-      valueServiceSpy.getValue.and.returnValue(stubValue);
+      const valueServiceSpy = {
+        getValue: vi.fn().mockReturnValue(stubValue),
+      };
```
we can probably assert that as `ValueService` as it's done elsewhere

AryanBagade replied:
Reverted to using jasmine.createSpyObj which doesn't need any type assertion at all.

AryanBagade committed:
docs: use agnostic async patterns in testing examples

Replace DoneFn callbacks with async/await and firstValueFrom for Observable tests. This makes the examples work with both Jasmine and Vitest while maintaining all existing Jasmine patterns.

WuglyakBolgoink commented on adev/src/content/examples/testing/src/app/demo/demo.spec.ts:
```diff
@@ -1,5 +1,6 @@
 // #docplaster
 import {LightswitchComponent, MasterService, ValueService, ReversePipe} from './demo';
+import {firstValueFrom} from 'rxjs';
```
```suggestion
import {firstValueFrom} from 'rxjs';
import {LightswitchComponent, MasterService, ValueService, ReversePipe} from './demo';
```

Import third party libs should be on first places, then internal imports.

AryanBagade committed:
Update adev/src/content/examples/testing/src/app/demo/demo.spec.ts

Co-authored-by: Ali Elderov <elderov@gmail.com>

Jyothireddy-pula reviewed (commented):
fine
